functions/New-DbaRgResourcePool.ps1

```diff
@@ -0,0 +1,197 @@
+
+    .NOTES
+        Tags: ResourcePool, ResourceGovernor
+        Author: John McCall (@lowlydba), https://www.lowlydba.com/
+
+        Website: https://dbatools.io
+        Copyright: (c) 2018 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/New-DbaRgResourcePool
+
+    .EXAMPLE
+        PS C:\> New-DbaRgResourcePool -SqlInstance sql2016 -ResourcePool "poolAdmin"
+
+        Creates a new resource pool named "poolAdmin" for the instance sql2016.
+
+    .EXAMPLE
+        PS C:\> New-DbaRgResourcePool -SqlInstance sql2012\dev1 -ResourcePool "poolDeveloper" -SkipReconfigure
+
+        Creates a new resource pool named "poolDeveloper" for the instance dev1 on sq2012.
+        Reconfiguration is skipped and the Resource Governor will not be able to use the new resource pool
+        until it is reconfigured.
+    #>
+
+    [CmdletBinding(SupportsShouldProcess, DefaultParameterSetName = "Default", ConfirmImpact = "Low")]
+    param (
+        [parameter(Mandatory, ValueFromPipeline, ValueFromPipelineByPropertyName)]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [string[]]$ResourcePool,
+        [ValidateSet("Internal", "External")]
+        [string]$Type = "Internal",
+        [Parameter(ParameterSetName = "Internal")]
+        [ValidateRange(0, 100)]
+        [int]$MinimumCpuPercentage,
+        [Parameter(Mandatory)]
+        [ValidateRange(1, 100)]
+        [int]$MaximumCpuPercentage,
+        [Parameter(Mandatory, ParameterSetName = "Internal")]
+        [ValidateRange(1, 100)]
+        [int]$CapCpuPercentage,
+        [Parameter(ParameterSetName = "Internal")]
+        [ValidateRange(0, 100)]
+        [int]$MinimumMemoryPercentage,
+        [Parameter(Mandatory)]
+        [ValidateRange(1, 100)]
+        [int]$MaximumMemoryPercentage,
+        [Parameter(ParameterSetName = "Internal")]
+        [ValidateRange(0, 2147483647)]
+        [int]$MinimumIOPSPerVolume,
+        [Parameter(ParameterSetName = "Internal")]
+        [ValidateRange(1, 2147483647)]
+        [int]$MaximumIOPSPerVolume,
+        [Parameter(ParameterSetName = "External")]
+        [int]$MaximumProcesses,
+        [switch]$SkipReconfigure,
+        [switch]$Force,
+        [switch]$EnableException
+    )
+    process {
+        foreach ($instance in $SqlInstance) {
+            try {
+                $server = Connect-DbaInstance -SqlInstance $instance -SqlCredential $SqlCredential
+            } catch {
+                Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+            }
+
+            foreach ($resPool in $ResourcePool) {
+                $existingResourcePool = Get-DbaRgResourcePool -SqlInstance $instance -Type $Type | Where-Object { $_.Name -eq $resPool }
```

Comment:
You should use `$server` instead of `$instance` here.
You should use `Where-Object Name -eq $resPool`.